localstack/services/dynamodb/server.py

```diff
@@ -106,8 +106,13 @@ def create_dynamodb_server(
     server = DynamodbServer(port)
     db_path = f"{config.dirs.data}/dynamodb" if not db_path and config.dirs.data else db_path
 
-    if is_env_true("DYNAMODB_IN_MEMORY"):
+    if is_env_true("DYNAMODB_IN_MEMORY") or not (
+        config.PERSISTENCE and os.environ.get("LOCALSTACK_API_KEY", "").strip()
```

Comment:
nit: We could re-use one of the existing util functions for this (e.g., `ls_ext.bootstrap.licensing.api_key_configured`). We could also introduce a new `is_advanced_persistence_enabled()` utility - this may in fact also be relevant for StepFunctions. /cc @giograno Not critical, though.

Reply:
Switched to `localstack.utils.persistence.is_persistence_enabled` and `localstack.utils.bootstrap.is_api_key_configured` (moved to community in #1155)